src/core_plugins/markdown_vis/public/markdown_vis_controller.js

```diff
@@ -1,21 +1,111 @@
 import MarkdownIt from 'markdown-it';
-import React from 'react';
+import React, { Component } from 'react';
 
 const markdownIt = new MarkdownIt({
   html: false,
   linkify: true
 });
 
-export function MarkdownVisComponent(props) {
-  const visParams = props.vis.params;
+/**
+ * The MarkdownVisComponent renders markdown to HTML and presents it.
+ */
+class MarkdownVisComponent extends Component {
+
+  constructor(props) {
+    super(props);
+    // Transform the markdown from parameters into HTML and store it in the state.
+    this.state = {
+      renderedMarkdown: this.transformMarkdown(props)
+    };
+  }
+
+  /**
+   * This method is used to actually render markdown from the passed parameter
```

Comment:
not sure if we need all this comments, but if we intend this to serve as a tutorial, then i am fine with it :)

Reply:
Yeah I think we can use it that way properly as a reference to show people how you can optimize your rendering performance. That's why I even put in some React explanations, that are not very specific to Markdown.